File: ios/ChildrenViewManager.m
#import "ChildrenViewManager.h"
#import "ChildrenView.h"

@implementation ChildrenViewManager

RCT_EXPORT_MODULE(ChildrenView)

RCT_EXPORT_VIEW_PROPERTY(uuid, NSString*)

- (UIView *)view
{
  return [ChildrenView new];
}

@end

File: android/src/main/java/com/callstack/reactnativebrownfield/ChildrenViewManager.kt
package com.callstack.reactnativebrownfield

import android.widget.FrameLayout
import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.annotations.ReactProp


class ChildrenViewManager : SimpleViewManager<FrameLayout>() {
    companion object {
        val REACT_CLASS = "ChildrenView"
    }

    override fun getName(): String {
        return REACT_CLASS
    }

    override fun createViewInstance(context: ThemedReactContext): FrameLayout {
        return FrameLayout(context)
    }

    @ReactProp(name = "uuid")
    fun setUuid(view: FrameLayout, uuid: String) {
        view.addView(ReactNativeBrownfield.shared.getRegisteredView(uuid))
    }
}


